client/rest-high-level/src/main/java/org/elasticsearch/client/eql/EqlSearchRequest.java

```diff
@@ -172,14 +175,26 @@ public EqlSearchRequest implicitJoinKeyField(String implicitJoinKeyField) {
         return this;
     }
 
+    public int size() {
+        return this.size;
+    }
+
+    public EqlSearchRequest size(int size) {
+        this.size = size;
+        if (fetchSize <= 0) {
+            throw new IllegalArgumentException("size must be greater than 0");
+        }
+        return this;
+    }
+
     public int fetchSize() {
         return this.fetchSize;
     }
 
     public EqlSearchRequest fetchSize(int size) {
```

Comment:
I would rename the arg to `fetchSize`

Reply:
Will rename it in a follow-up PR. I'd like to merge this in considering it took 3 times due to transient issues and backport it to 7.x